File: CommonCode/LoggingUtilities.m
//
//  LoggingUtilities.m
//  Spectrogeddon
//
//  Created by Tom York on 16/06/2016.
//  Copyright © 2016 Random. All rights reserved.
//

#import "LoggingUtilities.h"
#import <mach/mach_time.h>

void DLOGTime()
{
    DLOGEventTime(nil);
}

void DLOGEventTime(NSString* event)
{
#ifndef NDEBUG
    mach_timebase_info_data_t timebaseInfo = {0, 0};
    mach_timebase_info(&timebaseInfo);
    const uint64_t timeInAbs = mach_absolute_time();
    const uint64_t timeInNanoseconds = (timebaseInfo.numer == 1 && timebaseInfo.denom == 1) ? timeInAbs : (timebaseInfo.numer * timeInAbs / timebaseInfo.denom);
    NSLog(@"%@: %@", event, @(timeInNanoseconds));
#endif
}

void DLOGEventOnThread(NSString* event)
{
#ifndef NDEBUG
    const NSUInteger threadHash = [[NSThread currentThread] hash];
    NSLog(@"%@: %@", event, @(threadHash));
    
#endif
}

File: Spectrogeddon/Core/Source/LoggingUtilities.swift
//
//  LoggingUtilities.swift
//  SpectrogeddonOSX
//
//  Created by Tom York on 26/05/2019.
//  Copyright © 2019 Spectrogeddon. All rights reserved.
//

import Foundation

func DLOGTime() {
    DLOGEventTime(event: nil)
}

func DLOGEventTime(event: String?) {
    
    #if !NDEBUG
    var timebaseInfo = mach_timebase_info_data_t(numer: 0, denom: 0)
    mach_timebase_info(&timebaseInfo)
    let timeInAbs = mach_absolute_time()
    let timeInNanoseconds: UInt64
    if timebaseInfo.numer == 1 && timebaseInfo.denom == 1 {
        timeInNanoseconds = timeInAbs
    } else {
        timeInNanoseconds = UInt64(timebaseInfo.numer) * timeInAbs / UInt64(timebaseInfo.denom)
    }
    NSLog("\(event ?? "<unknown>"): \(timeInNanoseconds)")
    #endif

}

func DLOGEventOnThread(event: String?) {
    
    #if !NDEBUG
    let threadHash = Thread.current.hash
    NSLog("\(event ?? "<unknown>"), \(threadHash)")
    #endif
}
